src/metabase/query_processor/util.clj

```diff
@@ -127,7 +127,8 @@
   ensure survives."
   [fresh pre-existing]
   (let [by-key (m/index-by (comp field-ref->key :field_ref) pre-existing)]
-    (for [{:keys [field_ref] :as col} fresh]
-      (if-let [existing (get by-key (field-ref->key field_ref))]
+    (for [{:keys [field_ref source] :as col} fresh]
```

Comment:
Need to update it here because we merge metadata with the dataset metadata at the end.
https://github.com/metabase/metabase/blob/d93e15b5a0e511cb978c157b3b140466f6d3eb16/src/metabase/query_processor/middleware/annotate.clj#L674